drivers/dac/dac_dacx0501.c

```diff
@@ -0,0 +1,201 @@
+				   sizeof(raw_data));
+	if (status) {
+		return status;
+	}
+
+	/* DAC is big endian. */
+	*data = sys_get_be16(raw_data);
+	return 0;
+}
+
+static int dacx0501_reg_write(const struct device *dev, uint8_t addr, uint16_t data)
+{
+	const struct dacx0501_config *config = dev->config;
+	uint8_t write_cmd[3] = {addr};
+
+	/* DAC is big endian. */
+	sys_put_be16(data, write_cmd + 1);
+
+	return i2c_write_dt(&config->i2c_spec, write_cmd, sizeof(write_cmd));
+}
+
+static int dacx0501_channel_setup(const struct device *dev,
+				  const struct dac_channel_cfg *channel_cfg)
+{
+	const struct dacx0501_config *config = dev->config;
+	struct dacx0501_data *data = dev->data;
+	uint16_t device_id;
+	int status;
+
+	/* DACx0501 series only has a single output channel. */
+	if (channel_cfg->channel_id != 0) {
+		LOG_ERR("%s: Unsupported channel %d", __func__, channel_cfg->channel_id);
+		return -ENOTSUP;
+	}
+	if (data->configured) {
+		LOG_ERR("%s: Channel already configured", __func__);
+		return -EINVAL;
+	}
+	if (!i2c_is_ready_dt(&config->i2c_spec)) {
+		LOG_ERR("%s: I2C bus %s not ready", __func__, config->i2c_spec.bus->name);
+		return -ENODEV;
+	}
+	status = dacx0501_reg_read(dev, DACX0501_REG_DEVICE_ID, &device_id);
+	if (status) {
+		LOG_ERR("%s: read DEVICE_ID register failed", __func__);
+		return status;
+	}
+
+	/* See DEVICE_ID register RES field in the data sheet. */
+	data->resolution = 16 - 2 * FIELD_GET(DACX0501_MASK_DEVICE_ID_RES, device_id);
+
+	if (channel_cfg->resolution != data->resolution) {
+		LOG_ERR("%s: Unsupported resolution %d. Actual: %d", __func__,
+			channel_cfg->resolution, data->resolution);
+		return -ENOTSUP;
+	}
+
+	status = dacx0501_reg_write(
+		dev, DACX0501_REG_CONFIG,
+		FIELD_PREP(DACX0501_MASK_CONFIG_REF_PWDWN, config->ref_source == REF_EXTERNAL));
+	if (status) {
+		LOG_ERR("%s: write CONFIG register failed", __func__);
+		return status;
+	}
+	status = dacx0501_reg_write(
+		dev, DACX0501_REG_GAIN,
+		FIELD_PREP(DACX0501_MASK_GAIN_REFDIV_EN, config->voltage_mult == VM_DIV2) |
+			FIELD_PREP(DACX0501_MASK_GAIN_BUFF_GAIN, config->voltage_mult == VM_MUL2));
+	if (status) {
+		LOG_ERR("%s: GAIN Register update failed", __func__);
+		return status;
+	}
+
+	data->configured = true;
+
+	return 0;
+}
+
+static int dacx0501_write_value(const struct device *dev, uint8_t channel, uint32_t value)
+{
+	struct dacx0501_data *data = dev->data;
+
+	if (channel != 0) {
+		LOG_ERR("dacx0501: Unsupported channel %d", channel);
+		return -ENOTSUP;
+	}
+
+	if (!data->configured) {
+		LOG_ERR("dacx0501: Channel not initialized");
+		return -EINVAL;
+	}
+
+	if (value >= (1 << data->resolution)) {
+		LOG_ERR("dacx0501: Value %d out of range", value);
+		return -EINVAL;
+	}
+
+	value <<= (16 - data->resolution);
+
+	return dacx0501_reg_write(dev, DACX0501_REG_DAC, value);
+}
+
+static int dacx0501_init(const struct device *dev)
+{
+	struct dacx0501_data *data = dev->data;
+
+	data->configured = false;
```

Comment:
Not necessary. False is the default for static boolean variables anyway. You can remove this function and set it to `NULL` below.

Reply:
Done